crates/hyperqueue/src/dashboard/ui/screens/home/screen.rs

```diff
@@ -1,45 +1,47 @@
 use termion::event::Key;
 
-use crate::dashboard::ui::screen::Screen;
+use crate::dashboard::ui::screen::{change_to_worker_overview_screen, Screen, ScreenController};
 use crate::dashboard::ui::screens::home::cluster_overview_chart::ClusterOverviewChart;
 use crate::dashboard::ui::screens::home::worker_utilization_table::WorkerUtilTable;
 use crate::dashboard::ui::styles::style_header_text;
 use crate::dashboard::ui::terminal::DashboardFrame;
 use crate::dashboard::ui::widgets::text::draw_text;
 
 use crate::dashboard::data::DashboardData;
-use crate::dashboard::ui::screens::home::worker_info_table::WorkerInfoTable;
 use tui::layout::{Constraint, Direction, Layout, Rect};
 
 #[derive(Default)]
-pub struct HomeScreen {
+pub struct ClusterOverviewScreen {
     worker_util_table: WorkerUtilTable,
-    worker_info_table: WorkerInfoTable,
     cluster_overview: ClusterOverviewChart,
 }
 
-impl Screen for HomeScreen {
+impl Screen for ClusterOverviewScreen {
     fn draw(&mut self, frame: &mut DashboardFrame) {
         let layout = HomeLayout::new(frame);
         draw_text("HQ top", layout.header_chunk, frame, style_header_text());
 
-        self.cluster_overview.draw(layout.chart_chunk, frame);
-        self.worker_util_table.draw(layout.body_chunk, frame);
-        self.worker_info_table.draw(layout.info_chunk, frame);
+        self.cluster_overview.draw(layout.worker_count_chunk, frame);
+        self.worker_util_table
+            .draw(layout.worker_util_table_chunk, frame);
     }
 
-    fn update(&mut self, data: &DashboardData) {
+    fn update(&mut self, data: &DashboardData, _controller: &mut ScreenController) {
         self.worker_util_table.update(data);
         self.cluster_overview.update(data);
-        self.worker_info_table
-            .update(data, self.worker_util_table.get_selected_item());
     }
 
     /// Handles key presses for the components of the screen
-    fn handle_key(&mut self, key: Key) {
+    fn handle_key(&mut self, key: Key, controller: &mut ScreenController) {
         match key {
             Key::Down => self.worker_util_table.select_next_worker(),
             Key::Up => self.worker_util_table.select_previous_worker(),
+            Key::Right => {
+                change_to_worker_overview_screen(
```

Comment:
This would be more natural as a method on `ScreenController` I think. Also the method should receive worker ID, not option (you shouldn't be able to switch to the screen without providing the ID). Therefore it should only be enabled if some ID (worker) is selected.